pwiz_tools/Shared/Common/DataBinding/Controls/Editor/DocumentationViewer.cs

```diff
@@ -17,31 +18,216 @@
+            {
+                // Execute JavaScript to get the document's HTML content
+                var task = webView2.CoreWebView2.ExecuteScriptAsync(@"document.documentElement.outerHTML");
+                CommonActionUtil.RunAsync(() =>
+                {
+                    try
+                    {
+                        var encodedHtml = task.Result;
+                        // Decode Unicode escape sequences (e.g., \u003C -> <)
+                        var decodedHtml = System.Text.RegularExpressions.Regex.Unescape(encodedHtml);
+                        // Remove leading and trailing quotation marks from JavaScript string
+                        var rawHtml = decodedHtml.Trim('"');
+                        // Normalize line endings to match WebView2's normalization
+                        _webView2Html = rawHtml;
+                    }
+                    catch (Exception ex)
+                    {
+                        // Log cleanup errors for debugging but don't throw
+                        System.Diagnostics.Debug.WriteLine($@"Failed to get WebView2 outerHtml: {ex}");
+                    }
+                });
+            }
+            catch (Exception ex)
+            {
+                // Log cleanup errors for debugging but don't throw
+                System.Diagnostics.Debug.WriteLine($@"Failed to execute script on WebView2: {ex}");
+            }
+            return string.Empty;
+        }
+
+        private CoreWebView2Environment InitWebView2Environment()
+        {
+            if (string.IsNullOrEmpty(TestWebView2EnvironmentDirectory))
+                return null;
+
+            // If a test environment directory is specified, use it
+            _testDataFolder = TestWebView2EnvironmentDirectory;
+            Directory.CreateDirectory(_testDataFolder);
+                    
+            // Create environment with custom user data folder and additional options for test isolation
+            var options = new CoreWebView2EnvironmentOptions();
+            // Force WebView2 to use the test directory for all temp files and cache
+            options.AdditionalBrowserArguments = CommonTextUtil.SpaceSeparate(
+                $@"--user-data-dir=""{_testDataFolder}""",
+                $@"--disk-cache-dir=""{_testDataFolder}\\cache\""",
+                @" --disk-cache-size=0",
+                @"--disable-web-security",
+                @"--disable-features=VizDisplayCompositor",
+                @"--no-sandbox",
+                @"--disable-gpu",
+                @"--disable-background-timer-throttling",
+                @"--disable-backgrounding-occluded-windows",
+                @"--disable-renderer-backgrounding");
+            
+            // Create environment with custom user data folder on UI thread
+            return CoreWebView2Environment.CreateAsync(null, _testDataFolder, options).Result;
+        }
+
+        private void RunUI(Action act)
+        {
+            Invoke(act);

```

Comment:
I would recommend using "CommonActionUtil.SafeBeginInvoke" here because it handles the case where the form has already been closed.
If it were important that you use "Invoke" instead of "BeginInvoke" then you would not want to use SafeBeginInvoke, but I don't think we care in this case.

Reply:
Sure. I also changed the function name to RunUIAsync to be clearer about what it is doing.